### PR TITLE
docs: update readme and refactor tests

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
## Issue number

- Updated docs with heredoc
- Updated tests to work with new validation rules
- go deps update

## Checklist before requesting a review
- [x] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [x] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

**Comment by matt-FFFFFF** [comment]
Hey @lonegunmanb I have changed some of the validation rules, I then had to update the tests :)

I removed the validation rule where the hub_virtual_networks map key must equal the name. We might have a scenario where the vnet name is based on a value from the dependency graph, therefore not knows until apply.

File: variables.tf [hunk 1]
```diff
@@ -1,76 +1,113 @@
...
+    - `name` - The name of the delegation.
+    - `service_delegation` - An object with the following fields:
+      - `name` - The name of the service delegation.
+      - `actions` - A list of actions that should be delegated, the list is specific to the service being delegated.
+
+DESCRIPTION
   type = map(object({
     name                = string
-    address_spaces      = list(string)
+    address_space       = list(string)
```
Comment (lonegunmanb) [inline variables.tf:56]: Will `address_space` be considered as one string?

**Review by lonegunmanb — COMMENTED** [review]
Almost LGTM but only one question @matt-FFFFFF

File: variables.tf (hunk 1 above)
Comment (matt-FFFFFF) [reply variables.tf:56]: I was following the AzureRM precedent, though I have just noticed that subnets input is also using the plural.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_network.html#address_space

Comment (matt-FFFFFF) [reply variables.tf]: back to plural

**Comment by lonegunmanb** [comment]
It looks like we have some Terraform syntax erros @matt-FFFFFF

**Comment by matt-FFFFFF** [comment]
Thanks I'll sort today

**Review by lonegunmanb — APPROVED** [review]
Thanks @matt-FFFFFF, LGTM, I'll submit a new pr to fix e2e, let's just merge this one first.

**Pull request merged**